src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -8,7 +8,7 @@
 def load_arguments(self, _):
     from argcomplete.completers import FilesCompleter
 
-    from azure.mgmt.resource.resources.models import DeploymentMode
+    from azure.mgmt.resource.resources.models import DeploymentMode, WhatIfResultFormat
```

Comment:
Please use get_models() to import model.

Reply:
Done. I didn't changed the two imports below though. I didn't find any ResourceType enum definition for `managedapplications`. Loading it using get_models also failed a number of tests. Since I'm not familiar with the code I'll keep them as is.

Reply:
BTW may I also ask a question about get_models? So when can the `resource_type` be omitted? I searched the code base - it appears that sometimes you need to explicitly pass a ResourceType enum and sometimes you don't.

Reply:
when the resource type is the same as the one in __init__  as shown here https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/__init__.py#L19

Reply:
Got it. Thanks!